src/sentry/static/sentry/app/components/button.jsx

```diff
@@ -0,0 +1,59 @@
+import jQuery from 'jquery';
+import React from 'react';
+import classNames from 'classnames';
+
+import LoadingIndicator from '../components/LoadingIndicator';
+
+const Button = React.createClass({
+  propTypes: {
+    progress: React.PropTypes.bool,
+    disabled: React.PropTypes.bool,
+    icon: React.PropTypes.string
+  },
+
+  getDefaultProps() {
+    return {
+      progressBar: false,
+      disabled: false,
+    };
+  },
+  getInitialState() {
+    return {
+      isInitiated: false,
+      isComplete: false,
+    };
+  },
+
+  click() {
+    this.setState({isInitiated: true});
+    setTimeout(this.setState.bind(this, {isComplete: true}), 500);
```

Comment:
@mattrobenolt sorry if it wasn't clear but this is WIP. the timers are just there so I can see the effect until there is other data to drive it with.